src/redis/pool.cr

```diff
@@ -0,0 +1,37 @@
+require "pool/connection"
+
+class Redis::Pool
+  getter pool
+
+  def initialize(*args, pool = 5, pool_timeout = 5.0, **args2)
+    @pool = ConnectionPool(Redis).new(capacity: pool, timeout: pool_timeout) do
+      Redis.new(*args, **args2)
+    end
+  end
+
+  macro method_missing(call)
```

Comment:
I find this API confusing, where I make Redis command calls on the pool instance. I took me a while to understand what is happening here, because the connection pools I have used so far have an API like ysbaddaden/pool - `checkout`/`checkout` and `connection |conn| do ... end`.

I think a better API would be separate class that offers this functionality of transparently talking to the connection pool.

Like this:

```Crystal
redis_pool = Redis::Pool.new(host: ...)
# Returns a Redis::ClientProxy with a reference to redis_pool.
redis = redis_pool.client_proxy
# Redis::ClientProxy gives you the `method_missing` magic that transparently handles connections.
redis.set("mike", "rules")
redis.get("mike").should eq("rules")
```

No need to checkin a client proxy. And the change on the API surface is minimal - just an extra call to `redis_pool.client_proxy`.

Reply:
i not sure if redis_pool is usefull anywhere. may be just rename this class to `pooled_client = Redis::PooledClient`, so there would be no confusing. if you want access to pool, you can call  `pooled_client.pool.checkout`

Reply:
I like the idea of just renaming this class to `Redis::PooledClient`!

```Crystal
redis = Redis::PooledClient.new(host: ..., pool_size: ...)
redis.set("mike", "rules")
redis.get("mike").should eq("rules")
```